triple-reads-backend/index.ts

```diff
@@ -201,6 +211,183 @@ app.get('/books/:isbn', async (req: Request, res: Response) => {
   }
 });
 
+//POST TO INSERT
+app.post('/books', async (req: Request, res: Response) => {
+  try {
+    const { title, isbn, datePublished, abstract, image, authorNames, genreNames, publisher, admin, dateAdded } = req.body;
+
+    const adminString = await getAdminID(admin);
+
+    let authorString = "schema:author";
+
+    for (let i = 0; i < authorNames.length; i++) {
+
+      if (i !== authorNames.length - 1) {
+        authorString += " ex:" + authorNames[i].replace(/\s+/g, '_') + ",";
+      }
+      else {
+        authorString += " ex:" + authorNames[i].replace(/\s+/g, '_') + " ;";
+      }
+
+      const authorQuery = `
+        PREFIX schema: <http://schema.org/>
+        PREFIX ex: <http://example.org/>
+
+        INSERT DATA {
+          ex:${authorNames[i].replace(/\s+/g, '_')} a schema:Person ;
+            schema:name "${authorNames[i]}" .
+        }
+      `;
+
+      addAuthor(authorQuery);
+    }
+
+    let genreString = "schema:genre";
+
+    for (let i = 0; i < genreNames.length; i++) {
+
+      if (i !== genreNames.length - 1) {
+        genreString += " ex:" + genreNames[i].replace(/\s+/g, '_') + ",";
+      }
+      else {
+        genreString += " ex:" + genreNames[i].replace(/\s+/g, '_') + " ;";
+      }
+
+      const genreQuery = `
+        PREFIX schema: <http://schema.org/>
+        PREFIX ex: <http://example.org/>
+
+        INSERT DATA {
+          ex:${genreNames[i].replace(/\s+/g, '_')} a ex:Genre ;
+            schema:name "${genreNames[i]}" .
+        }
+      `;
+
+      addGenre(genreQuery);
+    }
+
+
+
+    const bookQuery = `
+      PREFIX schema: <http://schema.org/>
+      PREFIX ex: <http://example.org/>
+
+      INSERT DATA {
+        ex:${isbn} a schema:Book ;
+          schema:title "${title}" ;
+          schema:isbn "${isbn}" ;
+          schema:datePublished "${datePublished}" ;
+          schema:abstract "${abstract}" ;
+          schema:image "${image}" ;
+          ex:addedBy ex:${adminString} ;
+          schema:dateAdded "${dateAdded}" ;
+          ${authorString}
+          ${genreString}
+          schema:publisher "${publisher}" .
+      }
+    `;
+
+    await axios.post(fusekiUrlUpdate, null, {
+      params: {
+        update: bookQuery,
+      },
+    });
+
+    res.status(201).json({ message: 'Book inserted successfully' });
+  } catch (error) {
+    console.error('Error inserting book:', error);
+    res.status(500).json({ error: 'Error inserting book' });
+  }
+});
+
+app.delete('/books/:isbn', async (req, res) => {
+  try {
+    const isbnToDelete = req.params.isbn;  
+
+    const query = `
+      PREFIX schema: <http://schema.org/>
+      PREFIX ex: <http://example.org/>
+
+      DELETE WHERE {
+        ?book a schema:Book ;
+          schema:isbn "${isbnToDelete}" ;
+          ?p ?o .
+      }
+    `;   
+
+    await axios.post(fusekiUrlUpdate, null, {
+      params: {
+        update: query,
+      },
+    });
+
+    res.status(201).json({ message: 'Book deleted successfully' });
```

Comment:
This should first check if the isbn number actually exists. If the isbn number does not exist, then the status code should be 404 (Book not found).